api/org/commcare/api/persistence/UserDatabaseHelper.java

```diff
@@ -0,0 +1,357 @@
+            preparedStatement.setInt(1, id);
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+
+    public static ResultSet selectFromTable(Connection c, String storageKey, String[] fields, String[]values, Persistable p){
+        TableBuilder mTableBuilder = new TableBuilder(storageKey);
+        mTableBuilder.addData(p);
+        Pair<String, String[]> mPair = createWhere(fields, values, p);
+
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement("SELECT * FROM " + storageKey + " WHERE " + mPair.first +";");
+            for(int i=0; i<mPair.second.length; i++){
+                preparedStatement.setString(i+1, mPair.second[i]);
+            }
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs;
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    public static int insertToTable(Connection c, String storageKey, Persistable p){
+
+        Pair<String, List<Object>> mPair = getTableInsertData(storageKey, p);
+
+        try {
+            PreparedStatement preparedStatement = c.prepareStatement(mPair.first);
+            for(int i=0; i<mPair.second.size(); i++){
+                Object obj = mPair.second.get(i);
+
+                if(obj instanceof String){
+                    preparedStatement.setString(i + 1, (String) obj);
+                } else if(obj instanceof Blob){
+                    preparedStatement.setBlob(i+1, (Blob) obj);
+                } else if(obj instanceof Integer){
+                    preparedStatement.setInt(i + 1, ((Integer) obj).intValue());
+                } else if(obj instanceof byte[]){
+                    preparedStatement.setBinaryStream(i+1,new ByteArrayInputStream((byte[]) obj), ((byte[]) obj).length);
+                }
+            }
+            int affectedRows = preparedStatement.executeUpdate();
+
+            if (affectedRows == 0) {
+                throw new SQLException("Creating user failed, no rows affected.");
+            }
+
+            try (ResultSet generatedKeys = preparedStatement.getGeneratedKeys()) {
+                if (generatedKeys.next()) {
+                    int id = generatedKeys.getInt(1);
+                    p.setID(id);
+                    return id;
+                }
+                else {
+                    throw new SQLException("Creating user failed, no ID obtained.");
+                }
+            }
+
+        } catch (SQLException e) {
+            System.out.println("e: " + e);
+            e.printStackTrace();
+        }
+        return -1;
+    }
+
+    public static byte[] getRecordForValue(Connection c, String storageKey, String id){
+        String sqlStatement = "SELECT " + DATA_COL + " from " + storageKey + " WHERE commcare_sql_id = " + id + ";";
```

Comment:
This constructor should still be in here, but rest of method shoudl move to jdbc